pkg/phlaredb/head_queriers.go

```diff
@@ -142,6 +143,14 @@ func (q *headOnDiskQuerier) MergeByLabels(ctx context.Context, rows iter.Iterato
 	return seriesByLabels.normalize(), nil
 }
 
+func (q *headOnDiskQuerier) Series(ctx context.Context, params *ingestv1.SeriesRequest) ([]*typesv1.Labels, error) {
```

Comment:
As of now we keep TSDB in memory until the block is flushed on disk, therefore the head does not store any labels/series on disk. So I think we could return an empty response here

Reply:
Fixed in https://github.com/grafana/pyroscope/pull/2431/commits/d671a6a2ed6b0e7efb027ec052aa646d972d7fef